streams/src/main/java/org/apache/kafka/streams/processor/internals/PartitionGroup.java

```diff
@@ -49,44 +46,46 @@ public TopicPartition partition() {
             return queue.partition();
         }
 
-        public RecordQueue queue() {
+        RecordQueue queue() {
             return queue;
         }
     }
 
     // since task is thread-safe, we do not need to synchronize on local variables
     private int totalBuffered;
 
-    public PartitionGroup(Map<TopicPartition, RecordQueue> partitionQueues, TimestampExtractor timestampExtractor) {
-        this.queuesByTime = new PriorityQueue<>(partitionQueues.size(), new Comparator<RecordQueue>() {
+    PartitionGroup(final Map<TopicPartition, RecordQueue> partitionQueues) {
+        queuesByTime = new PriorityQueue<>(partitionQueues.size(), new Comparator<RecordQueue>() {
 
             @Override
-            public int compare(RecordQueue queue1, RecordQueue queue2) {
-                long time1 = queue1.timestamp();
-                long time2 = queue2.timestamp();
-
-                if (time1 < time2) return -1;
-                if (time1 > time2) return 1;
+            public int compare(final RecordQueue queue1, final RecordQueue queue2) {
+                final long time1 = queue1.timestamp();
```

Comment:
Of course one can argue! :) Readability is subjective and the sweet spot between the amount of information presented versus conciseness varies from person to person (and it often changes over time for the same person). Verbosity is additional noise and makes it both harder to understand what the code is doing _and_ makes it easier to miss important details.

To give a concrete example, I (and all the other reviewers) missed a bug introduced in a recent Streams PR because a removed line got lost in the noise of other clean-ups.

Finally, `final` only says that the variable doesn't change. In Java, collections are usually mutable, so mutability is ever present.